src/Spice86.Core/Emulator/OperatingSystem/DosFileManager.cs

```diff
@@ -602,119 +556,57 @@ public class DosFileManager {
+        DateTime creationLocalDate = creationZonedDateTime.ToLocalTime();
+        DateTime creationLocalTime = creationZonedDateTime.ToLocalTime();
+        dosDiskTransferArea.FileDate = ToDosDate(creationLocalDate);
+        dosDiskTransferArea.FileTime = ToDosTime(creationLocalTime);
+        dosDiskTransferArea.FileSize = (ushort)attributes.Length;
+        dosDiskTransferArea.FileName = Path.GetFileName(matchingFile);
     }
 
-    private string? ToCaseSensitiveFileName(string? caseInsensitivePath) {
-        if (string.IsNullOrWhiteSpace(caseInsensitivePath)) {
-            return null;
+    /// <summary>
+    /// Creates a directory on disk.
+    /// </summary>
+    /// <param name="directory">The directory name to create</param>
+    /// <returns></returns>
+    /// <returns>A <see cref="DosFileOperationResult"/> with details about the result of the operation.</returns>
+    public DosFileOperationResult CreateDirectory(string directory) {
+        string hostPath = _dosPathResolver.PrefixWithHostDirectory(directory);
+
+        string? fullPath = _dosPathResolver.ToHostCaseSensitiveFullName(directory, true);
+        if (string.IsNullOrWhiteSpace(fullPath)) {
+            return FileNotFoundError(directory);
         }
 
-        string fileToProcess = ConvertUtils.ToSlashPath(caseInsensitivePath);
-        string? parentDir = Path.GetDirectoryName(fileToProcess);
-        if (File.Exists(fileToProcess) || Directory.Exists(fileToProcess) ||
-            (string.IsNullOrWhiteSpace(parentDir) == false && Directory.Exists(parentDir) && Directory.GetDirectories(parentDir).Length == 0)) {
-            // file exists or root reached, no need to go further. Path found.
-            return caseInsensitivePath;
+        string parentFolder = _dosPathResolver.GetFullNameForParentDirectory(fullPath);
+
+        if (!Directory.Exists(parentFolder)) {
+            CreateDirectory(_dosPathResolver.GetHostRelativePathToCurrentDirectory(parentFolder));
         }
 
-        string? parent = ToCaseSensitiveFileName(parentDir);
-        if (parent == null) {
-            // End of recursion, root reached
-            return null;
+        if (_dosPathResolver.IsThereAnyDirectoryOrFileWithTheSameName(directory, new DirectoryInfo(parentFolder))) {
+            return FileNotFoundError(directory);
         }
 
-        // Now that parent is for sure on the disk, let's find the current file
         try {
-            string? fileNameOnFileSystem = GetActualCaseForFileName(caseInsensitivePath);
-            if (string.IsNullOrWhiteSpace(fileNameOnFileSystem) == false) {
-                return fileNameOnFileSystem;
-            }
-            Regex fileToProcessRegex = FileSpecToRegex(Path.GetFileName(fileToProcess));
-            if (Directory.Exists(parent)) {
-                return Array.Find(Directory
-                    .GetFiles(parent), x => fileToProcessRegex.IsMatch(x));
-            }
+            Directory.CreateDirectory(hostPath);
```

Comment:
This may fail because hostPath is not case sensitive right?

Reply:
This is why IsThereAnyDirectoryOrFileWithTheSameName checks if we can create the folder first.

Reply:
Resolved outdated conversation.